src/OpenFin/openfin.ts

```diff
@@ -361,11 +361,11 @@ export class OpenFinContainer extends WebContainerBase {
             newOptions.name = Guid.newGuid();
         }
 
-        const newWindow = this.wrapWindow(new this.desktop.Window(newOptions));
-        this.emit("window-created", { sender: this, name: "window-created", window: newWindow, windowId: newOptions.name, windowName: newOptions.name });
-        Container.emit("window-created", { name: "window-created", windowId: newOptions.name, windowName: newOptions.name });
-        ContainerWindow.emit("window-created", { name: "window-created", windowId: newOptions.name, windowName: newOptions.name });
-        return newWindow;
+        return this.wrapWindow(new this.desktop.Window(newOptions, win => {
+            this.emit("window-created", { sender: this, name: "window-created", windowId: newOptions.name, windowName: newOptions.name });
+            Container.emit("window-created", { name: "window-created", windowId: newOptions.name, windowName: newOptions.name });
+            ContainerWindow.emit("window-created", { name: "window-created", windowId: newOptions.name, windowName: newOptions.name });
+        }));
```

Comment:
Events need to be fired upon the callback from new fin.desktop.Window as it might not exist yet if we do not wait